### ISSUE TITLE
Rewrite find dialog

### ISSUE THREAD

**Issue by trypsynth** [trypsynth/paperback#21]
The find dialog should be rewritten. We should stop using wxFindDialog and instead use our own home-grown dialog with regular expression support, better accessibility, less hacks, etc.

**Comment by trypsynth** [comment]
Suggestion from @a-singer on mastodon, documenting it here: up/down arrow should go through your previously searched find entries. I see no real need to store this in the config and think it should only stay as long as the app is open, but welcome thoughts.

**Comment by trypsynth** [comment]
This find dialog is far from done, but it now has all the features of the old find dialog plus history, so I'm calling this good enough to close. Regular expression support is planned for the future too.
